iModelCore/GeomLibs/serialization/src/FlatBuffer/FixedStructs.cpp

```diff
@@ -2011,10 +2018,10 @@ bvector<IGeometryPtr> *invalidGeometry)
     if (invalidGeometry)
         invalidGeometry->clear ();
     auto fbRoot = flatbuffers::GetRoot <BGFB::VariantGeometry>(fbStart);
-    if (nullptr != fbRoot
-        && fbRoot->has_geometry_type ()
-        && fbRoot->has_geometry ()
-        )
+    auto myVerifier = flatbuffers::Verifier(fbStart, buffer.size() - s_prefixBufferSize);
+    if (!fbRoot->Verify(myVerifier))
+        return false;

```

Comment:
Move the new verifier lines below, inside the if (nullptr != fbRoot && ... ) scope.